src/views/beans.ts

```diff
@@ -85,3 +90,27 @@ class BeansDataProvider implements vscode.TreeDataProvider<Bean | LiveProcess> {
 }
 
 export const beansProvider = new BeansDataProvider();
+
+export async function openBeanHandler(bean: Bean) {
+    // TODO: extract logic in sts.java.javadocHoverLink into jdtls as a utility
+    if (!bean.type) {
+        const details = await getBeanDetail(bean.processKey, bean.id);
+        if (details && details.length > 0) {
+            bean = { ...bean, ...details[0] };
+        }
+    }
+
+    if (bean.type) {
+        const bindingKey = `L${bean.type.replace(/\./g, "/")};`;
```

Comment:
Format type name. 
E.g. from `org.springframework.samples.petclinic.owner.VisitController` to `Lorg.springframework.samples.petclinic.owner.VisitController;`

Reply:
Is there any possibility that bean is a generic type, array type or nested type? If that might happen, will the conversion here still work?

Reply:
It probably won't work. That's why I comment here to emphasize it. They way STS gets `bindingKey` is as below:

https://github.com/spring-projects/sts4/blob/da16b399f3669f5f615f523ec54410cc9b2a7622/headless-services/commons/commons-java/src/main/java/org/springframework/ide/vscode/commons/jandex/BindingKeyUtils.java#L31-L37

The ideal solution is to extract utilities into jdtls, as we discussed before.